File: ios/main/StallionMetaConstants.m
//
//  StallionMetaConstants.m
//  DoubleConversion
//
//  Created by Jasbir Singh Shergill on 28/01/25.
//

#import "StallionMetaConstants.h"

@implementation StallionMetaConstants

+ (SwitchState)switchStateFromString:(NSString *)value {
    if ([value isEqualToString:@"PROD"]) return SwitchStateProd;
    if ([value isEqualToString:@"STAGE"]) return SwitchStateStage;
    @throw [NSException exceptionWithName:@"InvalidSwitchStateException" reason:@"Invalid SwitchState" userInfo:nil];
}

+ (SlotStates)slotStateFromString:(NSString *)value {
    if ([value isEqualToString:@"NEW_SLOT"]) return SlotStateNewSlot;
    if ([value isEqualToString:@"STABLE_SLOT"]) return SlotStateStableSlot;
    if ([value isEqualToString:@"DEFAULT_SLOT"]) return SlotStateDefaultSlot;
    return SlotStateDefaultSlot;
}

+ (NSString *)stringFromSwitchState:(SwitchState)state {
    switch (state) {
        case SwitchStateProd: return @"PROD";
        case SwitchStateStage: return @"STAGE";
        default: return @"unknown";
    }
}

+ (NSString *)stringFromSlotState:(SlotStates)state {
    switch (state) {
        case SlotStateNewSlot: return @"NEW_SLOT";
        case SlotStateStableSlot: return @"STABLE_SLOT";
        case SlotStateDefaultSlot: return @"DEFAULT_SLOT";
        default: return @"unknown";
    }
}

@end

File: android/src/main/java/com/stallion/storage/StallionMetaConstants.java
package com.stallion.storage;

public class StallionMetaConstants {

  // Enum for SwitchState
  public enum SwitchState {
    PROD, STAGE;

    public static SwitchState fromString(String value) {
      try {
        return SwitchState.valueOf(value.toUpperCase());
      } catch (IllegalArgumentException e) {
        throw new IllegalArgumentException("Invalid SwitchState: " + value);
      }
    }
  }

  // Enum for SlotStates
  public enum SlotStates {
    NEW_SLOT, STABLE_SLOT, DEFAULT_SLOT;

    public static SlotStates fromString(String value) {
      try {
        return SlotStates.valueOf(value.toUpperCase());
      } catch (IllegalArgumentException e) {
        throw new IllegalArgumentException("Invalid SlotStates: " + value);
      }
    }
  }
}
